ZlobinSV/lab_3/main.cpp

```diff
@@ -0,0 +1,237 @@
+public:
+    int* data;
+    int size;
+    int capacity;
+    MyVector() {
+        data = nullptr;
+        size = 0;
+        capacity = 0;
+    }
+    MyVector(int newsize) {
+        size = newsize;
+        capacity = size * 2;
+        data = new int[capacity];
+    }
+    ~MyVector() {
+        delete[] data;
+    }
+    void push_back(int val) {
+        if (size == 0){
+             size = 1;
+            capacity = size * 2;
+            data = new int[capacity];
+         data[0]=val;
+         }
+        else {
+        if (size < capacity) {
+            data[size] = val;
+            size++;
+        }
+        else {
+            capacity = size * 2;
+            int* tmp = data;
+            data = new int[capacity];
+            for (int i = 0; i < size; i++) {
+                data[i] = tmp[i];
+            }
+            data[size] = val;
+            size++;
+            delete[] tmp;
+        }
+        }
+    }
+    void insert(int num, int val) {
+        if (num < size) {
+            if (size < capacity) {
+                int* tmp = data;
+                data = new int[capacity];
+                for (int i = 0; i < num; i++) {
+                    data[i] = tmp[i];
+                }
+                data[num] = val;
+                size++;
+                for (int i = num+1; i < size; i++) {
+                    data[i] = tmp[i-1];
+                }
+                delete[] tmp;
+            }
+            else {
+                capacity = size * 2;
+                int* tmp = data;
+                data = new int[capacity];
+                for (int i = 0; i < num; i++) {
+                    data[i] = tmp[i];
+                }
+                data[num] = val;
+                size++;
+                for (int i = num+1; i < size; i++) {
+                    data[i] = tmp[i - 1];
+                }
+                delete[] tmp;
+            }
+        }
+        else if(num == size){
+            push_back(val);
+        }
+        else { cout << "Error, out of vector" << endl; }
+    }
+    void delete_a(int num) {
+        if (num < size) {
+                int* tmp = data;
+                data = new int[capacity];
+                for (int i = 0; i < num; i++) {
+                    data[i] = tmp[i];
+                }
+                for (int i = num+1; i < size; i++) {
+                    data[i-1] = tmp[i];
+                }
+                size--;
+                delete[] tmp;
+        }
+        else { cout << "Error, out of vector" << endl; }
+    }
+    int* get_reference(int num) {
+        if (num < size) {
+            
+            return (data+num);
+        }
+        else {
+            cout << "Error, out of vector" << endl; return nullptr;
+        }
+    }
+    int get_size() {
+        return size;
+    }
+    void print_data() {
+        for (int i = 0; i < size; i++) {
+            cout << data[i] << "  ";
+        }
+        cout << endl;
+    }
+};
+
+class MyList {
```

Comment:
MyList must be class which encapsulated all logic related to linked list operations, such as modification, retrieving data and others.